content/articles/2019-10-dials-0-0-3.Rmarkdown

```diff
@@ -0,0 +1,74 @@
+---
+title: dials 0.0.3
+date: 2019-10-01
+slug: dials-0-0-3
+author: Max Kuhn
+categories: [package]
+description: >
+    A major update to the tuning parameter package. 
+photo:
+  url: https://unsplash.com/photos/QfN0l0MZCyc
+  author: Adi Goldstein
+---
+
+A new version of `dials` is [on CRAN](https://cran.r-project.org/package=dials). The package has contains basic frameworks for managing tuning parameters for models. It is a significant update to the package. The major change is that parameter objects are now generated by functions (as opposed to the prototype objects in the previous version). For example, to make a `dials` object for the number of PCA components in a model: 
+
+```{r pca, eval = FALSE}
+# previously
+pca_comps <- num_comp
+
+# now
+pca_comps <- num_comp()
+```
+
+For numeric parameters, the range of values can be set using the first argument:
+
+```{r pca-range}
+library(tidymodels)
+num_comp()
+
+num_comp(range = c(2, 10))
+```
+
+Sets of tuning parameters can be created and managed:
+
+```{r p-set}
+boosting_set <- param_set(list(trees(), splits = tree_depth(), min_n()))
+boosting_set
+
+# modifying the parameter range:
+boosting_set %>% update(trees = trees(c(100, 1000)))
+```
+
+Note that the tree depth parameter has a user-defined identification variable. This cna come in handy when there are multiple tuning parameters of the same type. For example, suppose two variables (`x1` and `x2`) were modeled using splines. The flexibility of each grouped be represented in a parameter set:
```

Comment:
```suggestion
Note that the tree depth parameter has a user-defined identification variable. This can come in handy when there are multiple tuning parameters of the same type. For example, suppose two variables (`x1` and `x2`) were modeled using splines. The flexibility of each grouped be represented in a parameter set:
```

Reply:
I was... um.. just checking to see if anyone would notice that. 😐